wgengine/magicsock/endpoint.go

```diff
@@ -475,6 +476,14 @@ func (de *endpoint) send(buffs [][]byte) error {
 	var err error
 	if udpAddr.IsValid() {
 		_, err = de.c.sendUDPBatch(udpAddr, buffs)
+
+		// If the OS reports the address as unreachable, clear out any potential
+		// reasons that we decided to prefer this address, as they're no longer
+		// valid.
+		if errors.Is(err, unix.EHOSTUNREACH) {
```

Comment:
this allocs per call because `unix.EHOSTUNREACH` is a uintptr (not a pointer) so it can't go in an interface value.

You'll want to do `var errHOSTUNREACH error = unix.EHOSTUNREACH` as a global to do it once and then use that err. And maybe `if err != nil && errors.Is` too IIRC, if that still matters.